### ISSUE TITLE
fs2 upgrade 2.2.0 - ClassCastException

### ISSUE THREAD

**Issue by regis-leray** [functional-streams-for-scala/fs2#1745]
After upgrading `fs2:2.0.1` to `fs2:2.2.0` im getting this stacktrace, it looks there is some optimisation have been done with `Chunk` 

I tried with fs2:2.1.0 and everything work fines

```
[info] - should readFile *** FAILED ***
[info]   java.lang.ClassCastException: class [Ljava.lang.Object; cannot be cast to class [B ([Ljava.lang.Object; and [B are in module java.base of loader 'bootstrap')
[info]   at fs2.Chunk.toBytes(Chunk.scala:276)
[info]   at fs2.io.file.FileHandle$$anon$1.$anonfun$write$1(FileHandle.scala:148)
[info]   at scala.runtime.java8.JFunction0$mcI$sp.apply(JFunction0$mcI$sp.scala:17)
[info]   at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:87)
[info]   at cats.effect.internals.IORunLoop$.startCancelable(IORunLoop.scala:41)
[info]   at cats.effect.internals.IOBracket$BracketStart.run(IOBracket.scala:86)
[info]   at cats.effect.internals.Trampoline.cats$effect$internals$Trampoline$$immediateLoop(Trampoline.scala:70)
[info]   at cats.effect.internals.Trampoline.startLoop(Trampoline.scala:36)
[info]   at cats.effect.internals.TrampolineEC$JVMTrampoline.super$startLoop(TrampolineEC.scala:93)
[info]   at cats.effect.internals.TrampolineEC$JVMTrampoline.$anonfun$startLoop$1(TrampolineEC.scala:93)
```

if you want to see the whole code it is here =>
https://github.com/regis-leray/fs2-ftp/pull/11

**Comment by mpilquist** [comment]
Thanks for the report. This reproduces:

```scala
> Chunk(0, 0).map(_.toByte).toBytes
java.lang.ClassCastException: class [Ljava.lang.Object; cannot be cast to class [B ([Ljava.lang.Object; and [B are in module java.base of loader 'bootstrap')
  fs2.Chunk.toBytes(Chunk.scala:276)
```

**Comment by mpilquist** [comment]
Introduced by #1715. We could roll that back and release a 2.2.1. @CremboC any thoughts? Problem is that `map` is creating an `Array[Any]` but then later when `toArray` is called, it's cast to an `Array[Byte]`.

**Comment by CremboC** [comment]
I was worried that this could happen. Hoped that tests would catch this, but obviously was too optimistic! Could just deoptimise `map` back to what it was before, or just undo the whole PR. Up to you! Sorry about that.

**Comment by mpilquist** [comment]
Fixed by #1746
